packages/arb-util/configuration/configuration.go

```diff
@@ -0,0 +1,393 @@
+package configuration
+
+import (
+	"context"
+	"fmt"
+	"github.com/knadh/koanf"
+	"github.com/knadh/koanf/parsers/json"
+	"github.com/knadh/koanf/providers/confmap"
+	"github.com/knadh/koanf/providers/file"
+	"github.com/knadh/koanf/providers/posflag"
+	"github.com/mitchellh/mapstructure"
+	"github.com/offchainlabs/arbitrum/packages/arb-util/ethutils"
+	"github.com/pkg/errors"
+	"github.com/rs/zerolog/log"
+	flag "github.com/spf13/pflag"
+	"math/big"
+	"os"
+	"path"
+	"time"
+)
+
+var logger = log.With().Caller().Stack().Str("component", "configuration").Logger()
+
+type Bridge struct {
+	Utils struct {
+		Address string `koanf:"address"`
+	} `koanf:"utils"`
+}
+
+type FeedInput struct {
+	Timeout time.Duration `koanf:"timeout"`
+	URL     string        `koanf:"url"`
+}
+
+type FeedOutput struct {
+	Addr string `koanf:"addr"`
+	HTTP struct {
+		Timeout time.Duration `koanf:"timeout"`
+	} `koanf:"http"`
+	Port    string        `koanf:"port"`
+	Ping    time.Duration `koanf:"ping"`
+	Timeout time.Duration `koanf:"timeout"`
+	Queue   int           `koanf:"queue"`
+	Workers int           `koanf:"workers"`
+}
+
+type Feed struct {
+	Input  FeedInput  `koanf:"input"`
+	Output FeedOutput `koanf:"output"`
+}
+
+type Healthcheck struct {
+	Addr   string `koanf:"addr"`
+	Enable bool   `koanf:"enable"`
+	L1Node struct {
+		Enable bool `koanf:"enable"`
+	} `koanf:"l1-node"`
+	Metrics struct {
+		Enable bool   `koanf:"enable"`
+		Prefix string `koanf:"prefix"`
+	} `koanf:"metrics"`
+	Port      string `koanf:"port"`
+	Sequencer struct {
+		Enable bool `koanf:"enable"`
+	} `koanf:"sequencer"`
+}
+
+type Node struct {
+	Aggregator struct {
+		Inbox struct {
+			Address string `koanf:"address"`
+		} `koanf:"inbox"`
```

Comment:
I think this should be flattened to `inbox-address`/`InboxAddress` unless there's a chance we'll flatten this